poc/vdaf_mastic.py

```diff
@@ -326,11 +327,12 @@ def helper_proof_share(cls, flp_seed):
         )
 
     @classmethod
-    def joint_rand_part(cls, agg_id, flp_seed, beta_share, nonce):
+    def joint_rand_part(cls, agg_id, flp_seed, beta_share, public_share, nonce):
         return cls.Xof.derive_seed(
             flp_seed,
             cls.domain_separation_tag(USAGE_JOINT_RAND_PART),
-            byte(agg_id) + nonce + beta_share,
+            byte(agg_id) + nonce + beta_share +
+            cls.Vidpf.public_share_as_bytes(public_share),
```

Comment:
Better aligned with terminology from the VDAF draft
```suggestion
            cls.Vidpf.encode_public_share(public_share),
```